packages/admin/blocks-admin/src/blocks/VimeoVideoBlock.tsx

```diff
@@ -0,0 +1,112 @@
+import { Field, FieldContainer, FinalFormInput, FinalFormRadio, FinalFormSwitch } from "@comet/admin";
+import { Box, FormControlLabel } from "@mui/material";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { VimeoVideoBlockData, VimeoVideoBlockInput } from "../blocks.generated";
+import { BlocksFinalForm } from "../form/BlocksFinalForm";
+import { SelectPreviewComponent } from "../iframebridge/SelectPreviewComponent";
+import { useAdminComponentPaper } from "./common/AdminComponentPaper";
+import { createBlockSkeleton } from "./helpers/createBlockSkeleton";
+import { BlockCategory, BlockInterface } from "./types";
+
+const isValidVimeoIdentifier = (value: string) => {
+    const urlRegEx = /(https?:\/\/)?(www\.)?(player\.)?vimeo\.com\/?(showcase\/)*([0-9)([a-z]*\/)*([0-9]{6,11})[?]?.*/;
+    const idRegEx = /([0-9]{6,11})/;
+
+    const urlMatch = value.match(urlRegEx);
+    const idMatch = value.match(idRegEx);
+
+    return !!urlMatch || !!idMatch;
+};
+
+const validateIdentifier = (value?: string) => {
+    return value && isValidVimeoIdentifier(value) ? undefined : (
+        <FormattedMessage id="comet.blocks.vimeoVideo.validation" defaultMessage="Should be a valid Vimeo URL or identifier" />
+    );
+};
+
+export const VimeoVideoBlock: BlockInterface<VimeoVideoBlockData, VimeoVideoBlockData, VimeoVideoBlockInput> = {
+    ...createBlockSkeleton(),
+
+    name: "VimeoVideo",
+
+    displayName: <FormattedMessage id="blocks.vimeoVideo" defaultMessage="Video (Vimeo)" />,
+
+    defaultValues: () => ({ vimeoIdentifier: "", autoplay: false, showControls: true, loop: false, aspectRatio: "16X9" }),
+
+    category: BlockCategory.Media,
+
+    // !vimeoIdentifier to allow saving empty string
+    isValid: ({ vimeoIdentifier }) => !vimeoIdentifier || isValidVimeoIdentifier(vimeoIdentifier),
+
+    createPreviewState: (state, previewCtx) => {
+        return { ...state, autoplay: false, adminMeta: { route: previewCtx.parentUrl } };
+    },
+    definesOwnPadding: true,
+    AdminComponent: ({ state, updateState }) => {
+        const isInPaper = useAdminComponentPaper();
+
+        return (
+            <Box padding={isInPaper ? 3 : 0} pb={0}>
+                <SelectPreviewComponent>
+                    <BlocksFinalForm
+                        onSubmit={(newState) => {
+                            updateState({ ...state, ...newState });
+                        }}
```

Comment:
```suggestion
                        onSubmit={(values) => {
                            updateState(values);
                        }}
```

We changed that in #2121 for YouTubeVideoBlock, so it should be changed here.

Reply:
https://github.com/vivid-planet/comet/pull/2068/commits/d8aa9f8fe983fdeb15d2b62083118db16875a5f7